src/go/rpk/pkg/cli/cloud/mcp/mcp.go

```diff
@@ -209,3 +197,314 @@ func newStdioCommand(fs afero.Fs, p *config.Params) *cobra.Command {
 	cmd.Flags().BoolVarP(&allowDelete, "allow-delete", "", false, "Allow delete RPCs. Off by default")
 	return cmd
 }
+
+func newProxyCommand(fs afero.Fs, p *config.Params) *cobra.Command {
+	var clusterID, mcpServerID string
+	var install bool
+	var mcpClient string
+	cmd := &cobra.Command{
+		Use:   "proxy",
+		Short: "Proxy requests to a remote Redpanda Cloud MCP server",
+		Long: `Proxy MCP requests to a remote Redpanda Cloud MCP server.
+
+This command connects to a specific cluster and MCP server running in that cluster,
+then proxies MCP requests from stdio to the remote MCP server over HTTP.
+
+Use --install to configure the MCP client instead of serving stdio.`,
+		Args: cobra.NoArgs,
+		PreRunE: func(_ *cobra.Command, _ []string) error {
+			if install && mcpClient == "" {
+				return fmt.Errorf("--client flag is required when using --install")
+			}
+			return nil
+		},
+		Run: func(_ *cobra.Command, _ []string) {
+			cfg, err := p.Load(fs)
+			out.MaybeDie(err, "Failed to load config: %w", err)
+
+			// Start out with empty token, and use the maybeReloadToken function to update.
+			cl := publicapi.NewCloudClientSet(cfg.DevOverrides().PublicAPIURL, "")
+			var m sync.RWMutex
+			var tokenOK bool
+
+			maybeReloadToken := func() error {
+				cfg, err := p.Load(fs)
+				if err != nil {
+					return err
+				}
+
+				var authVir *config.RpkCloudAuth
+				yVir := cfg.VirtualRpkYaml()
+				pVir := yVir.Profile(yVir.CurrentProfile)
+				authVir = pVir.VirtualAuth()
+				if authVir == nil {
+					authVir = yVir.CurrentAuth() // must be non-nil; we always have a default virtual auth
```

Comment:
We have a few helper functions to avoid doing this in https://github.com/redpanda-data/redpanda/blob/dev/src/go/rpk/pkg/config/rpk_yaml.go

Like `p.CurrentAuth()`